### PR TITLE
Update GitHub actions and Go to 1.19

### LINKED ISSUE #104: Bump go modules and CI to go v1.19
Per grooming, no objections on moving from go v1.18 to v1.19(latest stable). This requires two changes:

- go.mod
- CI(any Github action using a go runtime)

### PR DESCRIPTION
**Pull request by SaschaSchwarze0**
# Changes

Updating GitHub actions like in https://github.com/shipwright-io/build/pull/1085

Fixes #104

# Submitter Checklist

- [ ] Includes tests if functionality changed/was added
- [ ] Includes docs if changes are user-facing
- [x] [Set a kind label on this PR](https://prow.k8s.io/command-help#kind)  
- [x] Release notes block has been filled in, or marked NONE

# Release Notes

```release-note
NONE
```

### CODE REVIEW

File: .github/workflows/release.yaml [hunk 1]
```diff
@@ -22,14 +22,17 @@ jobs:
...
       with:
         fetch-depth: 0  # Fetch all history, needed for release note generation.
     # Install tools
-    - uses: actions/setup-go@v2
+    - name: Install Go
+      uses: actions/setup-go@v3
       with:
-        go-version: 1.18.x
-    - uses: sigstore/cosign-installer@v2.3.0
+        go-version: ${{ matrix.go-version }}
```
Comment (qu1queee) [inline .github/workflows/release.yaml:32]: seems not needed, could you keep it at 1.18, or maybe move it to 1.19 together with https://github.com/shipwright-io/operator/issues/104

Comment (SaschaSchwarze0) [reply .github/workflows/release.yaml]: Done.

**Review by HeavyWombat — COMMENTED** [review]
/lgtm
/approve

**Review by HeavyWombat — APPROVED** [review]

**Pull request merged**